File: src/uk/edu/populationfitness/test/IndividualTest.java
package uk.edu.populationfitness.test;

import org.junit.jupiter.api.Test;
import uk.edu.populationfitness.models.Config;
import uk.edu.populationfitness.models.Individual;

import static org.junit.jupiter.api.Assertions.*;

/**
 * Created by pete.callaghan on 03/07/2017.
 */
public class IndividualTest {

    @Test public void testOldGuyReadyToDie(){
        // Given an old individual
        Config config = new Config();
        int birth_year = 1964;
        int current_year = birth_year + config.max_age + 10;
        Individual individual = new Individual(config, birth_year);

        // Then they are ready to die
        assertTrue(individual.isReadyToDie(current_year));
    }

    @Test public void testYoungGuyNotReadyToDie(){
        // Given a young individual
        Config config = new Config();
        int birth_year = 1964;
        int current_year = birth_year + config.max_age - 10;
        Individual individual = new Individual(config, birth_year);

        // Then they are not ready to die
        assertFalse(individual.isReadyToDie(current_year));
    }

    @Test public void testCanBreed(){
        // Given a breeding age individual
        Config config = new Config();
        int birth_year = 1964;
        int current_year = birth_year + config.min_breeding_age + 1;
        Individual individual = new Individual(config, birth_year);

        // Then they can breed
        assertTrue(individual.canBreed(current_year));
    }

    @Test public void testYoungsterCannotBreed(){
        // Given a breeding age individual
        Config config = new Config();
        int birth_year = 1964;
        int current_year = birth_year + config.min_breeding_age - 1;
        Individual individual = new Individual(config, birth_year);

        // Then they can breed
        assertFalse(individual.canBreed(current_year));
    }

    @Test public void testOldsterCannotBreed(){
        // Given a breeding age individual
        Config config = new Config();
        int birth_year = 1964;
        int current_year = birth_year + config.max_breeding_age + 1;
        Individual individual = new Individual(config, birth_year);

        // Then they can breed
        assertFalse(individual.canBreed(current_year));
    }
}

File: DotNet/PopulationFitness/TestPopulationFitness/UnitTests/IndividualTest.cs
using PopulationFitness.Models;
using NUnit.Framework;

namespace TestPopulationFitness.UnitTests
{
    [TestFixture]
    public class IndividualTest
    {
        [TestCase]
        public void TSestOldGuyReadyToDie()
        {
            // Given an old individual
            Config config = new Config();
            int birth_year = 1964;
            int current_year = birth_year + config.MaxAge + 10;
            Epoch epoch = new Epoch(config, current_year);
            Individual individual = new Individual(epoch, birth_year);

            // Then they are ready to die
            Assert.True(individual.IsReadyToDie(current_year));
        }

        [TestCase]
        public void TestYoungGuyNotReadyToDie()
        {
            // Given a young individual
            Config config = new Config();
            int birth_year = 1964;
            int current_year = birth_year + config.MaxAge - 10;
            Epoch epoch = new Epoch(config, current_year);
            Individual individual = new Individual(epoch, birth_year);

            // Then they are not ready to die
            Assert.False(individual.IsReadyToDie(current_year));
        }

        [TestCase]
        public void TestCanBreed()
        {
            // Given a breeding age individual
            Config config = new Config();
            int birth_year = 1964;
            int current_year = birth_year + config.MinBreedingAge + 1;
            Epoch epoch = new Epoch(config, current_year);
            Individual individual = new Individual(epoch, birth_year);

            // Then they can breed
            Assert.True(individual.CanBreed(current_year));
        }

        [TestCase]
        public void TestYoungsterCannotBreed()
        {
            // Given a breeding age individual
            Config config = new Config();
            int birth_year = 1964;
            int current_year = birth_year + config.MinBreedingAge - 1;
            Epoch epoch = new Epoch(config, current_year);
            Individual individual = new Individual(epoch, birth_year);

            // Then they can breed
            Assert.False(individual.CanBreed(current_year));
        }

        [TestCase]
        public void TestOldsterCannotBreed()
        {
            // Given a breeding age individual
            Config config = new Config();
            int birth_year = 1964;
            int current_year = birth_year + config.MaxBreedingAge + 1;
            Epoch epoch = new Epoch(config, current_year);
            Individual individual = new Individual(epoch, birth_year);

            // Then they can breed
            Assert.False(individual.CanBreed(current_year));
        }
    }
}
